src/postamble.js

```diff
@@ -188,13 +190,19 @@ Module['callMain'] = function callMain(args) {
     var start = Date.now();
 #endif
 
+    var ret = Module[
 #if PROXY_TO_PTHREAD
-    // User requested the PROXY_TO_PTHREAD option, so call a stub main which pthread_create()s a new thread
-    // that will call the user's real main() for the application.
-    var ret = Module['_proxy_main'](argc, argv, 0);
+      // User requested the PROXY_TO_PTHREAD option, so call a stub main which pthread_create()s a new thread
+      // that will call the user's real main() for the application.
+      '_proxy_main'
 #else
-    var ret = Module['_main'](argc, argv, 0);
+      '_main'
 #endif
+    ](
+#if MAIN_READS_PARAMS
+      argc, argv
```

Comment:
This drops the 0, I assume that's fine? Presumably an undefined can get synthesized to a 0 in the case where we have a `char* envp[]` in main